File: LeetCode/Java/Easy/IntersectionOfTwoLinkedLists.java
package Easy;

public class IntersectionOfTwoLinkedLists {

	public static void main(String[] args){
		
	}
	
	
	public ListNode getIntersectionNode(ListNode headA, ListNode headB){
		int lenA = getLength(headA);
		int lenB = getLength(headB);
		
		if (lenA > lenB){
			int diff = lenA-lenB;
			while (diff > 0){
				headA = headA.next;
				diff--;
			}
		}
		if (lenB > lenA){
			int diff = lenB-lenA;
			while (diff > 0){
				headB = headA.next;
				diff--;
			}
		}
		
		while (headA!= null && headB != null){
			if (headA == headB){
				return headA;
			}
			headA = headA.next;
			headB = headB.next;
		}
		return null;
	}
	
	
	public int getLength(ListNode head){
		int count = 0;
		
		while (head != null){
			count++;
			head = head.next;
		}
		
		return count;
	}
	
}

File: LeetCode/Python/Easy/IntersectionOfTwoLinkedLists.py
class Solution(object):
    def getIntersectionNode(self, headA, headB):
        if headA is None or headB is None:
            return None

        lenHeadA = 1
        lenHeadB = 1
        ptrA = headA
        ptrB = headB

        while ptrA.next is not None or ptrB.next is not None:
            if ptrA.next is not None:
                lenHeadA += 1
                ptrA = ptrA.next
            if ptrB.next is not None:
                lenHeadB += 1
                ptrB = ptrB.next

        if ptrA != ptrB:
            return None

        ptrA = headA
        ptrB = headB
        while lenHeadA > lenHeadB:
            ptrA = ptrA.next
            lenHeadA -= 1
        while lenHeadB > lenHeadA:
            ptrB = ptrB.next
            lenHeadB -= 1

        while ptrA != ptrB:
            ptrA = ptrA.next
            ptrB = ptrB.next

        return ptrA
